src/sentry/api/endpoints/organization_member/__init__.py

```diff
@@ -28,28 +28,27 @@ def get_allowed_roles(
     organization: Organization,
     member: OrganizationMember | None = None,
 ) -> Collection[Role]:
+    """Get the set of roles that the request is allowed to manage.
+
+    In order to change another member's role, the returned set must include both
+    the starting role and the new role. That is, the set contains the roles that
+    the request is allowed to promote someone to and to demote someone from.
+    """
```

Comment:
The logic about checking both the starting role and the new role, and therefore gating demotions, lives elsewhere:

https://github.com/getsentry/sentry/blob/73bd578967fa96b3085ccbf79bae747672393ff7/src/sentry/api/endpoints/organization_member/details.py#L262-L267

However, the code snippet above enforces meaning on this function. So, I believe it calls for a docstring that spells it out as the function contract.

Although I refuse to apologize for ending a sentence with a preposition, I'm open to revisions if that last sentence of the docstring isn't easy enough to parse.